### DISCUSSION TITLE
How to create pop-up button using tailwindCSS className ?????

### DISCUSSION THREAD

**Discussion by ProPhycient5** [tailwindlabs/tailwindcss#7796] [Ideas]
I just simply want to create a button using TailwindCSS className so that, whenever I click on it, it must bounce only when u click, not that which tailwind animation provides, which keep on bouncing infinitely.

Plz do comment below your solution 👇🏼👇🏼👇🏼👇🏼👇🏼👇🏼

Looking forward for it.

**Comment by AviAvinav** [comment]
I think you will be able to do it with the "active" psuedo-class. You can have something like `className="active:animate-bounce"`. This way the animation will only activate when the button is pressed and stop when it is not pressed any longer. You can also check here in the docs: https://tailwindcss.com/docs/hover-focus-and-other-states#pseudo-classes

**Reply by ProPhycient5** [reply]
Thank you for the help. It is not bouncing smoothly while clicking on button.

**Reply by AviAvinav** [reply]
For making it smoother you can use classNames like transition, ease-in-out and duration-500. You can change the easing and duration as you like. Still, it doesn't make it very smooth when I have used it earlier, so, instead what you can do is not use the `animate-bounce` property and rather use something like this `className="active:scale-90"` and make it smoother with other properties. 

So, finally it would be something like `className="active:scale-90 transition duration-150 ease-out"`. You can see it in action here: https://play.tailwindcss.com/uR2H1JQU6n. Hope it helps. You can check the docs for more on the properties.

**Reply by ProPhycient5** [reply]
Yeah I have checked from the above link that button is working fine as I was wanting. But while using your same code, all the CSS stuffs are working except that `active:scale-90` style. I don't why is it not working ? 
For better understanding, I am attaching one gif where I have shown my code and that button on web page. 
Apart I am also attaching my `ts.config.js` file so that you will suggest me what more I need to add.

Gif 👇🏼
![btn-github](https://user-images.githubusercontent.com/71059909/157602559-e3f2257a-6aae-48a3-ad4e-6c62bdbec84a.gif)

My code on `ts.config.js` :
```
// tailwind.config.js
module.exports = {
  purge: {
    content: [
      "./pages/**/*.{js,ts,jsx,tsx}",
      "./components/**/*.{js,ts,jsx,tsx}",
    ],
    options: {
      safelist: [/(from|via|to|border|bg|text)-(.*)-(\\d{1}0{1,2})/],
    },
  },
  darkMode: false, // or 'media' or 'class'
  theme: {
    extend: {
      colors: {
        primary: "#2680eb",
        "dar-gray": "#4b4b4b",
        'sky-blue': '#007BFF',
        "light-gray-0": "#eaeaea",
        "light-gray-1": "rgb(75,75,75)",
        "light-gray-2": "rgb(128,128,128)",
        "dark-navy-blue": "#002643",
        "light-navy-blue": "#003A66",
        "medium-navy-blue": "#00467a",
        "color-picker": "#175787",
        "block-blue": "#014e87",
        "action-blue": "#005b9e",
        "night-blue": "#19183780",
        "renderer-gray": "rgb(224, 224, 224)",
        red: "#e34850",
        "green-400": "#2d9d78",
        "green-500": "#268e6c",
      },
      animation: {
        // If you are using Toast component
        left: "left 0.3s",
        right: "right 0.3s",
        // if you are using the animate variant of the modal
        modal: "modal 0.3s",
        // if you are using drawer variant right
        "drawer-right": "drawer-right 0.3s",
        // if you are using drawer variant left
        "drawer-left": "drawer-left 0.3s",
        // if you are using drawer variant top
        "drawer-top": "drawer-top 0.3s",
        // if you are using drawer variant bottom
        "drawer-bottom": "drawer-bottom 0.3s",
      },
      keyframes: {
        // if you are using drawer variant right
        "drawer-right": {
          "0%, 100%": { right: "-500px" },
          "100%": { right: "0" },
        },
        // if you are using drawer variant left
        "drawer-left": {
          "0%, 100%": { left: "-500px" },
          "100%": { left: "0" },
        },
        // if you are using drawer variant top
        "drawer-top": {
          "0%, 100%": { top: "-500px" },
          "100%": { top: "0" },
        },
        // if you are using drawer variant bottom
        "drawer-bottom": {
          "0%, 100%": { bottom: "-500px" },
          "100%": { bottom: "0" },
        },
        // if you are using the animate variant of the modal
        modal: {
          "0%, 100%": { top: "-500px" },
          "100%": { top: "0" },
        },
        /* If you are using Toast component*/
        left: {
          "0%, 100%": { transform: "translateX(-100%)" },
          "100%": { transform: "translateX(0)" },
        },
        right: {
          "0%, 100%": { transform: "translateX(100%)" },
          "100%": { transform: "translateX(0)" },
        },
      },
    },
    variants: {
      /* If you are using Collapse or Accordion component*/
      transitionProperty: {
        height: "height",
      },
    },
  },
  variants: {
    extend: {
      backgroundColor: ["active", "checked", "group-hover"],
      inset: ["checked"],
      opacity: ["disabled", "group-hover"],
      textColor: ["active", "visited", "first", "group-hover"],
      divideColor: ["group-hover"],
    },
  },
  plugins: [require("@tailwindcss/forms")],
};
```

**Reply by AviAvinav** [reply]
I think this is because you are using Tailwind CSS v2 and the code I shared was for v3 and active state is not enabled for any core plugins in v2, but no need to worry as you don't have to upgrade to v3 rather what you can do is, in your tailwind.config.js add this:

```
variants: {
      extend: {
         scale: ["active"],
       }
    }
```

complete tailwind.config.js:

```
// tailwind.config.js
module.exports = {
  purge: {
    content: [
      "./pages/**/*.{js,ts,jsx,tsx}",
      "./components/**/*.{js,ts,jsx,tsx}",
    ],
    options: {
      safelist: [/(from|via|to|border|bg|text)-(.*)-(\\d{1}0{1,2})/],
    },
  },
  darkMode: false, // or 'media' or 'class'
  theme: {
    extend: {
      colors: {
        primary: "#2680eb",
        "dar-gray": "#4b4b4b",
        'sky-blue': '#007BFF',
        "light-gray-0": "#eaeaea",
        "light-gray-1": "rgb(75,75,75)",
        "light-gray-2": "rgb(128,128,128)",
        "dark-navy-blue": "#002643",
        "light-navy-blue": "#003A66",
        "medium-navy-blue": "#00467a",
        "color-picker": "#175787",
        "block-blue": "#014e87",
        "action-blue": "#005b9e",
        "night-blue": "#19183780",
        "renderer-gray": "rgb(224, 224, 224)",
        red: "#e34850",
        "green-400": "#2d9d78",
        "green-500": "#268e6c",
      },
      animation: {
        // If you are using Toast component
        left: "left 0.3s",
        right: "right 0.3s",
        // if you are using the animate variant of the modal
        modal: "modal 0.3s",
        // if you are using drawer variant right
        "drawer-right": "drawer-right 0.3s",
        // if you are using drawer variant left
        "drawer-left": "drawer-left 0.3s",
        // if you are using drawer variant top
        "drawer-top": "drawer-top 0.3s",
        // if you are using drawer variant bottom
        "drawer-bottom": "drawer-bottom 0.3s",
      },
      keyframes: {
        // if you are using drawer variant right
        "drawer-right": {
          "0%, 100%": { right: "-500px" },
          "100%": { right: "0" },
        },
        // if you are using drawer variant left
        "drawer-left": {
          "0%, 100%": { left: "-500px" },
          "100%": { left: "0" },
        },
        // if you are using drawer variant top
        "drawer-top": {
          "0%, 100%": { top: "-500px" },
          "100%": { top: "0" },
        },
        // if you are using drawer variant bottom
        "drawer-bottom": {
          "0%, 100%": { bottom: "-500px" },
          "100%": { bottom: "0" },
        },
        // if you are using the animate variant of the modal
        modal: {
          "0%, 100%": { top: "-500px" },
          "100%": { top: "0" },
        },
        /* If you are using Toast component*/
        left: {
          "0%, 100%": { transform: "translateX(-100%)" },
          "100%": { transform: "translateX(0)" },
        },
        right: {
          "0%, 100%": { transform: "translateX(100%)" },
          "100%": { transform: "translateX(0)" },
        },
      },
    },
    variants: {
      /* If you are using Collapse or Accordion component*/
      transitionProperty: {
        height: "height",
      },
    },
  },
  variants: {
    extend: {
      backgroundColor: ["active", "checked", "group-hover"],
      inset: ["checked"],
      opacity: ["disabled", "group-hover"],
      textColor: ["active", "visited", "first", "group-hover"],
      divideColor: ["group-hover"],
      scale: ["active"],
    },
  },
  plugins: [require("@tailwindcss/forms")],
};
```

You can check the docs on this here:

https://v2.tailwindcss.com/docs/hover-focus-and-other-states#active
https://v2.tailwindcss.com/docs/scale

**Reply by ProPhycient5** [reply]
Thank you @AviAvinav for your observation and help 🙏🏼. But it's still not popping up, however, while hovering `active:scale-90` on code, I am see applied style on VSCode editor. So I have attached gif to have a look. 
So do I need to upgrade from older version ? 
If I upgrade, will it break previous CSS stuff that I have applied on the project?

Gif  👇🏼

![btn-github-2](https://user-images.githubusercontent.com/71059909/157640904-c5ae703c-8ce5-4eea-9803-283b79792b06.gif)

**Reply by AviAvinav** [reply]
I am glad I could be of help to you. As for why the animation is not working is because I forgot to add the transform className which was necessary for this in v2. Just add the `transform` className and you would be good to go. Here, I made this button in v2: https://play.tailwindcss.com/RTK5RIDREr.

As for upgrading to v3, you will have to look at the [upgrade guide](https://tailwindcss.com/docs/upgrade-guide). There are some new features, changes in the names of some classes and few color pallete changes, so depending on your project, you will have to see. Also, the structure of the config file has changed a little. Your project will work fine even with v2, upgrading an existing project, especially if its a big one, would be a bit of a hassle in my opinion.

Good Luck 👍

**Reply by ProPhycient5** [reply]
Thank you a lot. You are doing great job 🎉🎉🎉🎉👏👏👏👏 by serving the dev communities via open-source-contribution.
I am also quite new. I am also eager to contribute open-source and serve community. I have send connection on LinkedIn.
Do suggest me any projects concerned to `HTML, CSS, JS, ReactJS` so that I will start contribute actively.
Below is my linkedIn link👇🏼:
https://www.linkedin.com/in/sawan-kumar-das-b1511a1ab/
